README.md

```diff
@@ -44,8 +44,13 @@ There are two ways to specify AWS credentials in this connector:
 
    The connector will request a temporary token from the AWS STS service and assume a role from another AWS account.
    It requires `aws.sts.role.arn`, `aws.sts.role.session.name` to be specified.
+3) Use default provider chain or custom provider
+   
+    If you prefer to use AWS default provider chain, you can leave {`aws.access.key.id` and `aws.secret.access.key`} and
+    {`aws.sts.role.arn`, `aws.sts.role.session.name`} blank. In case you prefer to build your own custom
+    provider, pass the custom provider class as a parameter to `aws.credential.provider`
 
-It is important not to use both.
+It is important not to use both 1 and 2 simultaneously.
```

Comment:
Should it validate all 3? ex. if a custom credential provider is used (i.e. non default value) then, fail if sts or static credentials are provided.
This could help to avoid misconfigurations

Reply:
Yes, in order. Order is documented here. https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default

Reply:
Yes, in order. Reference is here: https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default